hummingbot/strategy/avellaneda_market_making/avellaneda_market_making.pyx

```diff
@@ -558,7 +558,10 @@ cdef class AvellanedaMarketMakingStrategy(StrategyBase):
                           f"    order_book_depth_factor(\u03BA)= {self._kappa:.5E}",
                           f"    volatility= {volatility_pct:.3f}%"])
             if self._time_left is not None:
-                lines.extend([f"    time until end of trading cycle= {str(datetime.timedelta(seconds=float(self._time_left)//1e3))}"])
+                if self._execution_timeframe == "infinite":
+                    lines.extend([f"    time until end of trading cycle= N/A"])
+                else:
+                    lines.extend([f"    time until end of trading cycle= {str(datetime.timedelta(seconds=float(self._time_left)//1e3))}"])
```

Comment:
Since we have an object that represents the execution type I believe we should delegate this decision to it (either directly or using double dispatch if we want the logic that builds the messages to still be in the strategy object). With this we are adding "if" decisions instead of using object oriented programming solutions.

Reply:
Also, if we let the execution state handle this, we don't need to force `process_tick` in the state to return a boolean